src/it/scala/temple/containers/DockerPostgresService.scala

```diff
@@ -0,0 +1,56 @@
+package temple.containers
+
+import java.sql.DriverManager
+
+import com.whisk.docker.{DockerCommandExecutor, DockerContainer, DockerContainerState, DockerKit, DockerReadyChecker}
+import temple.containers.DockerPostgresService.PostgresReadyChecker
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+
+/** DockerPostgresService encapsulates all configuration for running a postgres docker container */
+object DockerPostgresService {
+  val image            = "postgres:9.6"
+  val databaseUsername = "postgres"
+  val databasePassword = ""
+  val databaseName     = "temple_test"
+  val internalPort     = 5432
+  val externalPort     = 44444
+  val externalUrl      = s"jdbc:postgresql://localhost:$externalPort/$databaseName"
+
+  class PostgresReadyChecker extends DockerReadyChecker {
+
+    // Called when container has started
+    override def apply(
+      container: DockerContainerState
+    )(implicit docker: DockerCommandExecutor, ec: ExecutionContext): Future[Boolean] =
+      try {
+        val url = s"jdbc:postgresql://${docker.host}:$externalPort/"
+        Option(DriverManager.getConnection(url, databaseUsername, databasePassword))
+          .foreach { conn =>
+            try conn.createStatement().execute(s"CREATE DATABASE $databaseName")
+            finally conn.close()
+          }
+        Future.successful(true)
+      } catch {
+        case error: Throwable => Future.failed(error)
+      }
```

Comment:
```scala
val url = s"jdbc:postgresql://${docker.host}:$externalPort/"
Future {
  Option(DriverManager.getConnection(url, databaseUsername, databasePassword)) map { conn =>
    try conn.createStatement().execute(s"CREATE DATABASE $databaseName")
    finally conn.close()
  }
}

Reply:
Not sure this works - the type this returns is `Future[Unit]` not the required `Future[Boolean]`

Reply:
Add a line only containing true just before the Future closes